### PR TITLE
chore: solution templates generator (refs PPCDEV-12705)

### PR DESCRIPTION
**Pull request by ArekZc**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,3 +18,11 @@ jobs:
...
         - yarn build
         - yarn test
+      deploy:
+        provider: pages
+        local_dir: docs/build
+        skip_cleanup: true
+        github_token: $GITHUB_TOKEN  # Set in the settings page of your repository, as a secure variable
+        keep_history: true
+        on:
+          branch: PPCDEV-12705-generator
```
Comment (rezik007) [inline .travis.yml:28]: How about to create more generic branch for deployment? `deploy/release`?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```
+$ yarn build
+```
+
+This command generates static content into the `build` directory and can be served using any static contents hosting service.
+
+### Deployment
+
+```
+$ GIT_USER=<Your GitHub username> USE_SSH=true yarn deploy
```
Comment (rezik007) [inline docs/README.md:30]: How it is related to travis setup where deploy is triggered on given branch? If there are two ways it is worth to mention it here.

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+const generator = require('../templates');
+
+module.exports = {
+  title: 'Solution Templates',
```
Comment (rezik007) [inline docs/docusaurus.config.js:12]:
```suggestion
  title: 'Solutions Templates',
```

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  tagline: 'A library of detection helpers that can be used in Tag Managers to improve tracking abilities of Analytics software.',
+  url: 'https://piwikpro.github.io',
+  baseUrl: '/solutions-templates/',
+  onBrokenLinks: 'throw',
+  onBrokenMarkdownLinks: 'warn',
+  favicon: 'https://piwik.pro/wp-content/assets/img/favicon.png?ver=1608560712',
+  organizationName: 'Piwik PRO', // Usually your GitHub org/user name.
+  projectName: 'solution-templates', // Usually your repo name.
```
Comment (rezik007) [inline docs/docusaurus.config.js:20]:
```suggestion
  projectName: 'solutions-templates', // Usually your repo name.
```

File: docs/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  themeConfig: {
+    colorMode: {
+      disableSwitch: true,
+    },
+    navbar: {
+      title: 'Solution Templates',
```
Comment (rezik007) [inline docs/docusaurus.config.js:26]:
```suggestion
      title: 'Solutions Templates',
```

File: docs/package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "solution-templates",
```
Comment (rezik007) [inline docs/package.json:2]:
```suggestion
  "name": "solutions-templates",
```

File: docs/src/pages/index.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+function Home() {
+  const context = useDocusaurusContext();
+  const {siteConfig = {}} = context;
+
+  return (
+    <Layout
+      title={siteConfig.title}
+      description={siteConfig.tagline}>
+      <header className={clsx('hero hero--primary', styles.heroBanner)}>
```
Comment (rezik007) [inline docs/src/pages/index.js:105]: Why part of css are from module and some aren't?

File: docs/src/pages/index.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        <div className="container">
+          <h1 className="hero__title">{siteConfig.title}</h1>
+          <p className="hero__subtitle">{siteConfig.tagline}</p>
+        </div>
+      </header>
+      <main>
+        <section>
+          <div className="container">
```
Comment (rezik007) [inline docs/src/pages/index.js:113]: There is no such class 🤔  or is it from docusaurs "internal" stylesheets?

File: docs/src/pages/styles.module.css [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+  grid-area: label;
+}
+
+.annotatedLayoutBody {
+  grid-area: body;
+}
+
+.generateButton {
+  
+}
```
Comment (rezik007) [inline docs/src/pages/styles.module.css:132]: Empty ruleset

File: docs/package.json [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    "eslint-config-airbnb": "^18.0.1",
+    "eslint-config-prettier": "^6.7.0",
+    "eslint-plugin-header": "^3.0.0",
+    "eslint-plugin-import": "^2.21.2",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-react": "^7.20.0",
+    "eslint-plugin-react-hooks": "^4.0.4",
+    "http-server": "^0.12.3",
+    "prettier": "^2.0.2",
+    "stylelint": "^13.2.1"
```
Comment (rezik007) [inline docs/package.json:41]: `stylelint-copyright` is missing here.

File: docs/src/pages/styles.module.css (hunk 9 above)
Comment (rezik007) [reply docs/src/pages/styles.module.css:132]: Shouldn't stylelint catch that? 🤔

File: scripts/build.js [hunk 11]
```diff
@@ -4,36 +4,9 @@ const Mustache = require('mustache');
...
-
-    if (tag) {
-      data.runtime = `\n${tag.text}\n`;
-
-      fs.writeFileSync(path.join(__dirname, `../dist/${file}`), Mustache.render(template, data), {
-        encoding: 'utf-8',
-      });
-    }
-  }
+  execSync('yarn workspace solution-templates run build');
```
Comment (rezik007) [inline scripts/build.js:9]:
```suggestion
  execSync('yarn workspace solutions-templates run build');
```
if you would change workspace name as I proposed above

File: docs/README.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+This website is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.
+
+### Installation
+
+```
+$ yarn
+```
+
+### Local Development
```
Comment (rezik007) [inline docs/README.md:11]: To achieve that I had to:
- go to main dir and run yarn build
- then go to docs dir and run yarn start

File: docs/src/pages/index.js (hunk 7 above)
Comment (ArekZc) [reply docs/src/pages/index.js:105]: docusaurus already place some styles as theme if we want to use theme we need to use those classes as well

File: docs/src/pages/index.js (hunk 8 above)
Comment (ArekZc) [reply docs/src/pages/index.js:113]: internal docusaurus class

File: docs/src/pages/styles.module.css (hunk 9 above)
Comment (ArekZc) [reply docs/src/pages/styles.module.css:132]: there is no stylelint configure in this repository

File: docs/README.md (hunk 12 above)
Comment (ArekZc) [reply docs/README.md:11]: Yes, I need to update it

File: .travis.yml (hunk 1 above)
Comment (ArekZc) [reply .travis.yml:28]: just to test, deploy will be only from master

**Pull request merged**